src/AliasMacro.jl

```diff
@@ -15,7 +15,7 @@ as the two names really refer to the same object.
 The alias also gets a special docstring that points out that it is an alias
 
 # Examples
-```jldoctest; setup = :(using AbstractAlgebra)
+```jldoctest; setup = :(using AbstractAlgebra; AbstractAlgebra.set_current_module(@__MODULE__))
```

Comment:
I would prefer if this inline setup thing would not duplicate the content of `doctestsetup()` into this many files. Can't you call `doctestsetup()` instead here? (with some interpolation magic maybe)

Reply:
(I was actually wondering why we need to do this for every (?) docstring.)

Reply:
We don't use `setdocmeta!` because then Nemo and Oscar would start to break. This is explained at the opf of `docs/make.jl`, this PR slightly improves that explanation. Here it is for your convenience:
```
# We do *not* call `DocMeta.setdocmeta!` with value `:(using AbstractAlgebra)`
# because we want to force that each jldoctest has its individual
# `setup = :(using AbstractAlgebra)`.
# The reason is that the AbstractAlgebra doctests are run also in Nemo,
# where `using AbstractAlgebra` must be avoided, as a few symbols change
# their meaning, e.g. `QQ`.
```

If you inspect the final commit in this PR you'll see that I previously tried to do `setup = AbstractAlgebra.doctestsetup()` but it just didn't work.

Of course I'd also prefer if we could do something like this, or even better, use `setdocmeta!`, but any solution must keep the Nemo and Oscar doctests working.

But unless someone has a bright idea quickly, I'd still like to merge this as-is, for the improved doctest contents, and then I'll happily review someone else's review for resolving this in a better way :-)

Reply:
(E.g. perhaps it suffices to only apply the explicit setup block to doctests which reference `ZZ` or `QQ` or so. But I'd rather not spend more hours figuring that out just now)

Reply:
I wonder if this can be resolved later by changing Nemo to do 
```
DocMeta.setdocmeta!(Nemo, :DocTestSetup, :(using Nemo); recursive = true)
DocMeta.setdocmeta!(Nemo.AbstractAlgebra, :DocTestSetup, Nemo.AbstractAlgebra.doctestsetup(); recursive = true)
```

(And then similar in Hecke, Oscar). I'll open an issue as a reminder to figure this out.